src/utils/auth.rs

```diff
@@ -28,15 +32,155 @@ pub fn check_authorization_header(req: &ServiceRequest, expected_key: &SecretStr
             return false;
         }
 
-        return &SecretString::new(token) == expected_key;
+        return true;
     }
     false
 }
 
+/// Helper to validate API key and check permissions for global-scoped actions
+///
+/// Use this for endpoints that access resources globally (e.g., listing all relayers).
+/// All required permissions must be satisfied (AND logic).
+///
+/// # Arguments
+/// * `req` - The HTTP request containing the Authorization header
+/// * `api_key_repo` - Repository for looking up API keys
+/// * `required_actions` - List of required actions (e.g., ["relayers:read"])
+///
+/// # Returns
+/// `Ok(())` if the API key has all required permissions with global scope
+pub async fn validate_api_key_permissions(
+    req: &HttpRequest,
+    api_key_repo: &dyn ApiKeyRepositoryTrait,
+    required_actions: &[&str],
+) -> Result<(), ApiError> {
+    // Extract and validate Authorization header
+    let token = extract_token_from_request(req)?;
+
+    // Check if this is the environment API key (full access bypass)
```

Comment:
Potentially we should get from api key repo always to avoid some edge cases in the future. 

In memory mode repo is populated as well with env var value?